### DISCUSSION TITLE
Can't get labels to print as PDF

### DISCUSSION THREAD

**Discussion by chris-thorn** [inventree/InvenTree#4036] [Q&A] [answered]
Hello,

I cannot work out how to make stock location labels generate as PDFs. I've made some slight changes to the template HTML file, uploaded it in the admin interface and tested it in a browser and it works just fine as HTML. But I just can't seem to get my desktop browser (Chrome) to offer to download it as a PDF.

### Version Information:
InvenTree-Version: 0.9.0 dev
Django Version: 3.2.16
Commit Hash: 53ecca3
Commit Date: 2022-12-09
Database: postgresql
Debug-Mode: False
Deployed using Docker: True
Active plugins: False

This is what I have in the admin interface under **Label** > **Stock location labels**:

![image](https://user-images.githubusercontent.com/3541291/206708024-0f57241a-8171-496f-b4b0-d172aa481610.png)

The label preview I'm getting in my browser:

![image](https://user-images.githubusercontent.com/3541291/206708419-e3c1658e-8c58-4119-8c06-a622e78d4b42.png)

This is the code used to generate the label. Basically the same as the example, just with a few minor layout changes:

```
<style>
  @page {
    width: 60mm;
    height: 20mm;
    padding: 1mm;
    margin: 0px 0px 0px 0px;
  }

  .container {
    display: flex;
    flex-direction: row;
    font-size: 12pt;
  }

  .location-container {
    font-family: 'Noto Sans', Tahoma, sans-serif;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    display: flex;
    flex-direction: column;
    /*padding: 0.8mm 0;*/
    margin-left: 1mm;
  }

  .loc-name-container {
    display: flex;
    flex-grow: 1;
  }

  .loc-name {
    font-weight: bold;
  }

  .qr {
    width: 18mm;
    height: 18mm;
  }
</style>
{% load barcode %}
<div class="container">
  <img class='qr' src="{% qrcode location.format_barcode %}" />
  <div class='location-container'>
    <div class="loc-name-container">
      <span class="loc-name">{{ location.name }}</span>
    </div>
    <span class="loc-id">Location ID: <b>{{ location.id }}</b></span>
  </div>
</div>
```

Do I need to enable a specific plugin to make this work?

Thanks for reading!

Chris

**Comment by miggland** [comment, marked as answer]
There is a setting for a "debug" mode for reports which gives you HTML output instead of PDF. It applies to labels as well:

![image](https://user-images.githubusercontent.com/1314853/206711393-0206d867-5fa3-44e6-a1f4-5c1abff47f8c.png)

Check this setting. You may need to reload/empty your cache if it doesn't change.

**Reply by chris-thorn** [reply]
Aaargh thank you! I thought I had looked at every option, but I didn't notice the second 'Reporting' option in the list - I must have only looked at the first one!

All working now - I'll have a play. Thanks again!
